src/components/Trade/hooks/useSwapper/calculateAmounts.ts

```diff
@@ -30,18 +31,26 @@ export const calculateAmounts = ({
   sellAssetUsdRate,
   action,
   selectedCurrencyToUsdRate,
-  tradeFee,
+  sellAssetTradeFeeUsd,
+  buyAssetTradeFeeUsd,
 }: CalculateAmountsArgs): CalculateAmountsReturn => {
   const assetPriceRatio = bnOrZero(buyAssetUsdRate).dividedBy(sellAssetUsdRate)
   const usdAmount = bnOrZero(amount).dividedBy(selectedCurrencyToUsdRate)
   const cryptoSellAmount = toBaseUnit(usdAmount.dividedBy(sellAssetUsdRate), sellAsset.precision)
   const cryptoBuyAmount = toBaseUnit(usdAmount.dividedBy(buyAssetUsdRate), buyAsset.precision)
-  const tradeFeeBaseUnit = toBaseUnit(tradeFee, buyAsset.precision)
+  const sellAssetTradeFeeUsdBaseUnit = toBaseUnit(
+    sellAssetTradeFeeUsd.div(sellAssetUsdRate),
+    buyAsset.precision,
```

Comment:
Should be `sellAsset.precision`, tested with CoW swaps so that would explain why it was working since ETH and ERC20s (except USDc and a few others) have the same 18 precision

```suggestion
    sellAsset.precision,
```